### PR TITLE
CAMEL-11373: Create a Camel Twilio component

### PR DESCRIPTION
**Pull request by tadayosi**
https://issues.apache.org/jira/browse/CAMEL-11373

### CODE REVIEW

File: components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        }
+    }
+
+    /**
+     * What kind of operation to perform
+     *
+     * @return the API Name
+     */
+    public TwilioApiName getApiName() {
+
```
Comment (davsclaus) [inline components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConfiguration.java]: There is an empty line here

File: components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+@Converter
+public final class TwilioConverter {
+
+    private TwilioConverter() {
+        //Utility Class
+    }
+
+    @Converter
+    public static URI toURI(String value) throws URISyntaxException {
```
Comment (davsclaus) [inline components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java]: We should not have generic coverters like string -> uri in a component. They belong in camel-core etc. And what do you really need this for?

File: apache-camel/src/main/descriptors/common-bin.xml [hunk 3]
```diff
@@ -267,6 +267,7 @@
         <include>org.apache.camel:camel-test-spring</include>
         <include>org.apache.camel:camel-testng</include>
         <include>org.apache.camel:camel-tika</include>
+        <include>org.apache.camel:camel-twilio</include>
```
Comment (davsclaus) [inline apache-camel/src/main/descriptors/common-bin.xml]: You need to add camel-twillio-starter as well further below with the others

File: components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    @Converter
+    public static Account.Type toAccountType(String value) {
+        return Account.Type.forValue(value);
+    }
+
+    // -----------------------------------------------------
+    // com.twilio.rest.api.v2010.account
+    // -----------------------------------------------------
+
+    @Converter
```
Comment (davsclaus) [inline components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java]: I wonder if all these are not enum types? Camel has a generic string -> enum converter. Does it not work?

File: components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java (hunk 2 above)
Comment (tadayosi) [inline components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java]: The intention was to provide a way to convert String -> URI upon automatically setting `CamelTwilio.*` headers to the properties of a Twilio command object (API proxy) like we do in `camel-google-drive` component.

But you are right that it's not totally necessary here. Removing it.

File: components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java (hunk 4 above)
Comment (tadayosi) [inline components/camel-twilio/src/main/java/org/apache/camel/component/twilio/TwilioConverter.java]: Same as above. The generic converter just works fine. Removing them too.

**Comment by tadayosi** [comment]
Updated. Can you take a look again, please?

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by tadayosi** [comment]
Thanks @davsclaus!

**Comment by pilundain** [comment]
Where can can i find internal classes TwilioApiName and TwilioApiCollection?

**Pull request closed without merging**